src/armor_layers.cpp

```diff
@@ -112,387 +106,499 @@ item_penalties get_item_penalties( std::list<item>::const_iterator worn_item_it,
+                    }
+                    break;
+                    case column_divider: {
+                        if( is_selected ) { //fill whole line with color
+                            mvwprintz( w, point( right_bound, cur_print_y ), print_col, "%*s", col_w, "" );
+                        }
+                        nc_color print_col2 = is_selected ? hilite( c_light_gray ) : c_light_gray;
+                        mvwputch( w, point( right_bound + col_w - 1, cur_print_y ), print_col2, LINE_XOXO );
+                    }
+                    break;
+                    default: {
+                        int val = get_col_value( static_cast<sort_armor_columns>( col_idx ) );
+                        mvwprintz( w, point( right_bound, cur_print_y ), print_col, "%*d", col_w, val );
+                    }
+                    break;
+                }
+            }
+            nc_color name_col = penalties.color_for_stacking_badness();
+            if( is_selected ) {
+                name_col = hilite( name_col );
+            }
+            mvwprintz( w, point( 2, cur_print_y ), c_dark_gray, clothing_layer( itm ) );
+            if( is_selected ) { //fill whole line with color
+                mvwprintz( w, point( 9, cur_print_y ), name_col, "%*s", right_bound - 9, "" );
+            }
+            trim_and_print( w, point( 9, cur_print_y ), right_bound - 9, name_col, name );
+        }
+        void set_aggregate_data( std::array<int, column_num_entries> &agg ) {
+            for( size_t i = 0; i < column_num_entries; i++ ) {
+                if( i == column_warmth ) {
+                    agg[i] += get_col_value( static_cast<sort_armor_columns>( i ) );
+                } else {
+                    agg[i] += get_col_value( static_cast<sort_armor_columns>( i ) ) * get_col_value( column_coverage );
+                }
+            }
+        }
 };
 
-static std::vector<layering_item_info> items_cover_bp( const Character &c, int bp )
+class layering_group_info
 {
-    std::vector<layering_item_info> s;
-    for( auto elem_it = c.worn.begin(); elem_it != c.worn.end(); ++elem_it ) {
-        if( elem_it->covers( static_cast<body_part>( bp ) ) ) {
-            s.push_back( { get_item_penalties( elem_it, c, bp ),
-                           elem_it->get_encumber( c ),
-                           elem_it->tname()
-                         } );
+    private:
+        std::string name;
+        std::array<int, column_num_entries> header_data;
+        std::list<layering_item_info> items;
+        body_part bp;
+        player *p;
+        int start_line;
+
+        int divider_posn;
+
+        //printing constatnts
+        int x_bound;
+        int max_y;
+        //printing vars
+        int list_min_line;
+        int list_selected_line;
+        bool is_moving;
+
+    public:
+        bool is_skipped = false;
+
+        layering_group_info( body_part bp, player *p ) {
```

Comment:
```suggestion
        layering_group_info( body_part bp, player *p ) : bp( bp ), p( p ) { }
```
And the body is now empty. (It's call member initializer lists.)